### PR TITLE
Add info about pod and service CIDR while preparing AKS clusters for AKS-AKS clustermesh

### PR DESCRIPTION
**Pull request by amitmavgupta**
Please ensure your pull request adheres to the following guidelines:

- [ ] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [x] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [x] Thanks for contributing!

Fixes: #issue-number

```release-note
<!-- Enter the release note text here if needed or remove this section! -->
```

### CODE REVIEW

**Comment by amitmavgupta** [comment]
@PhilipSchmid FYI

File: Documentation/network/clustermesh/aks-clustermesh-prep.rst [hunk 1]
```diff
@@ -67,6 +67,9 @@ Install cluster one
     except for Azure reserved ranges and ranges used by other clusters we intend to
     add to the clustermesh.
 
+    .. note::
+        We need to set explicit values for pod-cidr and service-cidr else AKS will allocate (``10.244.0.0/16``) for pod-cidr and (``10.0.0.0/16``) for service-cidr.
+
```
Comment (learnitall) [inline Documentation/network/clustermesh/aks-clustermesh-prep.rst]:
```suggestion
    .. note::
        We need to set explicit values for pod CIDR and service CIDR, otherwise AKS will
        allocate ``10.244.0.0/16`` for pod CIDR and ``10.0.0.0/16`` for service CIDR.

```

**Review by learnitall — CHANGES_REQUESTED** [review]
Great note to add! I left a suggestion to help this change conform to our style guides, which can be found here: https://docs.cilium.io/en/stable/contributing/docs/docsstyle/. I adjusted the following:

* Remove hyphens
* Capitalize acronyms 
* Wrapped the line

Also this line implies that there would be negative consequences if we didn't set explicit values for the CIDRs, since you wrote that `we need to set` them. Is that true? If so, can you add a description of what the negative consequences would be? If not, then can you reword this to make it more clear that the values AKS sets are just default values, rather than incompatible values?

Thank you!

**Comment by amitmavgupta** [comment]
> Great note to add! I left a suggestion to help this change conform to our style guides, which can be found here: https://docs.cilium.io/en/stable/contributing/docs/docsstyle/. I adjusted the following:
> 
> * Remove hyphens
> * Capitalize acronyms
> * Wrapped the line
> 
> Also this line implies that there would be negative consequences if we didn't set explicit values for the CIDRs, since you wrote that `we need to set` them. Is that true? If so, can you add a description of what the negative consequences would be? If not, then can you reword this to make it more clear that the values AKS sets are just default values, rather than incompatible values?
> 
> Thank you!

@learnitall thanks. I have incorporated your feedback and squashed all commits into one and made the changes. Please let me know if that's ok or happy to make further changes as well.

Regards,
-Amit

File: bpf/bpf_lxc.c
Comment (YutaroHayakawa) [inline bpf/bpf_lxc.c:1]: Why do we need datapath changes here?

**Review by YutaroHayakawa — CHANGES_REQUESTED** [review]

File: bpf/bpf_lxc.c
Comment (amitmavgupta) [reply bpf/bpf_lxc.c:1]: @YutaroHayakawa not at all. Did my commit add that file change? I don't think I made any changes to bpf_lxc.

File: Documentation/network/clustermesh/aks-clustermesh-prep.rst [hunk 2]
```diff
@@ -67,6 +67,16 @@ Install cluster one
     except for Azure reserved ranges and ranges used by other clusters we intend to
     add to the clustermesh.
 
+    .. note::
+        We need to set explicit values for pod CIDR and service CIDR, otherwise AKS will
+        allocate ``10.244.0.0/16`` for pod CIDR and ``10.0.0.0/16`` for service CIDR.
+        These are default values that are assigned by AKS while creating a cluster if 
+        the user doesn't specify pod CIDR and service CIDR arguments while creating 
+        the AKS clusters.
```
Comment (learnitall) [inline Documentation/network/clustermesh/aks-clustermesh-prep.rst]: I think the last `while creating the AKS clusters` here is redundant, can you remove it?

```suggestion
        the user doesn't specify pod CIDR and service CIDR arguments.
```

**Review by learnitall — CHANGES_REQUESTED** [review]
Looks great! Thank you for making these changes! One more small nit, but then its a LGTM from me.

File: bpf/bpf_lxc.c
Comment (learnitall) [reply bpf/bpf_lxc.c:1]: Might need a rebase?
